MigrationGuide.md

```diff
@@ -82,14 +90,17 @@ connection.connect { error in
 }
 ```
 
-With the new method your code will need to be updated to handle the change in return type in a manner similar to that in the example below:
+Below is an example of the 3.0 API, using the QueryResult to check for an error:
 
 ```swift
 let connection = PostgreSQLConnection(….)
 // Connect the connection
 connection.connect { queryResult in
     guard let _ = queryResult.success else {
-        // Connection not established, call queryResult.asError to get error.
+        if let error = queryResult.asError {
+            return Log.error("Connection failed: \(error)")`
+        }
+        // Unknown error
```

Comment:
I think checking `queryResult.success` is redundant, checking `asError` is enough.  I'd simplify this to just the check for `asError`, otherwise you can safely assume success.  This goes for all the cases where you receive a `QueryResult` and don't need to inspect the result, other than knowing that it was successful.